### PR TITLE
Add GHC 9.4.1-rc1 hints

### PR DESCRIPTION
**Pull request by bergmark**

### CODE REVIEW

File: stack/global-hints.yaml [hunk 1]
```diff
@@ -1111,3 +1111,40 @@ ghc-7.10.1:
   pretty: 1.1.2.0
   template-haskell: 2.10.0.0
   directory: 1.2.2.0
+ghc-9.4.0.20220523:
```
Comment (mpilgrem) [inline stack/global-hints.yaml:1114]: Are you missing `Win32`?

File: stack/global-hints.yaml [hunk 2]
```diff
@@ -1111,3 +1111,40 @@ ghc-7.10.1:
...
+  ghc-bignum: 1.3
+  ghc-boot: 9.4.0.20220523
+  ghc-boot-th: 9.4.0.20220523
+  ghc-compact: 0.1.0.0
+  ghc-heap: 9.4.0.20220523
+  ghc-prim: 0.9.0
+  ghci: 9.4.0.20220523
+  haskeline: 0.8.2
+  hpc: 0.6.1.0
+  integer: gmp-1.1
```
Comment (mpilgrem) [inline stack/global-hints.yaml:1136]: Shoud this be `integer-gmp: 1.1`?

File: stack/global-hints.yaml (hunk 1 above)
Comment (bergmark) [reply stack/global-hints.yaml:1114]: Seems like it! I couldn't use `update-global-hints.hs` for the alpha, will check if i can find the Win32 version somewhere.

File: stack/global-hints.yaml (hunk 2 above)
Comment (bergmark) [reply stack/global-hints.yaml]: fixed

**Comment by mpilgrem** [comment]
@bergmark, is there something that I can compare your list to, to double-check that is complete and accurate? (The things I spotted previously were ad hoc observations). Also, is this overtaken by the release of ghc-9.4.1-rc1 (ghc-9.4.0.20220721)?

**Comment by bergmark** [comment]
update-global-hints.hs in this repo uses e.g. https://downloads.haskell.org/ghc/9.2.4/docs/html/users_guide/9.2.4-notes.html to find these versions, I asked [on discourse](https://discourse.haskell.org/t/ghc-9-4-1-rc1-is-now-available/4817/14?u=bergmark) if there is a better way to get this info.

There is https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history which I wouldn't necessarily trust as I think it's updated manually and there have been some inconsistencies in the past. Also the alphas are not mentioned there.

You can install the release yourself and run `ghc-pkg list`, which is what I did and how I missed Win32...

There is also `https://downloads.haskell.org/ghc/9.4.1-rc1/docs/libraries/index.html` which is missing e.g Win32.

File: stack/global-hints.yaml (hunk 1 above)
Comment (bergmark) [reply stack/global-hints.yaml]: Found it here https://downloads.haskell.org/ghc/9.4.1-rc1/docs/libraries/Win32-2.12.0.0/Graphics-Win32.html

**Comment by bergmark** [comment]
I *think* this fixes everything.

wrt verifying/testing, I was looking into whether I could fork stackage-content, but the URL is hard-coded into pantry so it'd be some work.

**Comment by mpilgrem** [comment]
I double-checked (on Windows, so could not check `unix-2.7.3`) by using Stack to install GHC and then `stack exec -- ghc-pkg list`.

(Installing with Stack was not straight-forward. For some reason, I kept getting:

~~~
Extracting ghc-9.4.0.20220721.tar...
Extracted total of 15205 files from ghc-9.4.0.20220721.tar
Problem while decompressing C:\Users\mikep\AppData\Local\Programs\stack\x86_64-windows\ghc-9.4.0.20220721-tmp-8c1332f48b729cbf\ghc-9.4.0.20220721.tar
~~~

but uncompressing manually with 7-Zip 19.0 worked fine.)

**Comment by mpilgrem** [comment]
@bergmark, it turned out we needed to quote versions that could be interpreted as numbers. See https://github.com/commercialhaskell/stack/issues/5819. I have done that.

**Comment by bergmark** [comment]
ouch! thanks for fixing!

**Pull request merged**